shared/src/pool_fetching.rs

```diff
@@ -655,13 +720,45 @@ mod tests {
         assert!(instance
             .fetch(hashset!(pair), BlockNumber::Latest)
             .await
+            .unwrap()
             .is_empty());
         // Change inner to return a new pool if it was called.
         pools.lock().await.push(Pool::uniswap(pair, (1, 1)));
         // Inner shouldn't get called because the previous empty result is still cached.
         assert!(instance
             .fetch(hashset!(pair), BlockNumber::Latest)
             .await
+            .unwrap()
             .is_empty());
     }
+
+    #[test]
+    fn pool_fetcher_forwards_node_error() {
+        let results = vec![FetchedPool {
+            reserves: Err(ethcontract_error::testing_node_error()),
+            pair: Default::default(),
+            token0_balance: Ok(1.into()),
+            token1_balance: Ok(1.into()),
```

Comment:
Isn't it also possible to get an eth contract error on either of these?

Reply:
Sure but I don't think it's worth it to repeat this test 3 times